app/js/services/DiseaseService.js

```diff
@@ -26,67 +42,68 @@ function DiseaseService($q, $resource, AppSettings, DiseaseModel, $log, filterFi
    *
    * @return {Array} filtered array of DiseaseModels
    */
-  service.query = (searchQuery, mutationsGenes)=>{
-    
-    
-      $log.log(`query:${AppSettings.api.baseUrl}${AppSettings.api.diseases}/`);
-
-
-      let diseasePromise = new Promise((resolve, reject)=>{
-        if ($localStorage.diseaseData && $localStorage.diseaseData.count) {
-          if($localStorage.diseaseData.count === 0) return  reject(`No Disease Types found matching: "${searchQuery}"`);
-          $log.log('fetch disease data from localStorage',$localStorage.diseaseData);
-           resolve($localStorage.diseaseData); 
-        }
-
-        $log.log('fetch disease data from DB resource');
-
-        DISEASES_RESOURCE.query((diseaseResponse) => {
+  service.query = (searchQuery, mutationsGenes) => {
+    $log.log(`query:${AppSettings.api.baseUrl}${AppSettings.api.diseases}/`);
+
+    let diseasePromise = new Promise((resolve, reject) => {
```

Comment:
If this is working well as is, it's probably fine to leave it, but you should consider switching from `Promise` to Angular's `$q.defer()` here (the API is slightly different, but easy to map over), to improve integration with the digest cycle. That might eliminate the need for the `$scope.$apply()` in `queryParamSelector.component.js`, line 124.

If you go that route, the `reject()` and `resolve()` lines below would be called off the closure variable returned by `$q.defer()`.